src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+import { getFilteredPeople } from '../../utils/getFilteredPeople';
+import { getSortedPeople } from '../../utils/getSortedPeople';
+import { SortBy } from '../../types/SortBy';
+import { SearchLink } from '../SearchLink';
+
+type Props = {
+  people: Person[],
+  personSlug: string,
+};
+
+export const PeopleTable: React.FC<Props> = React.memo(
+  ({ people, personSlug }) => {
+    const [searchParams] = useSearchParams();
+    const sort = searchParams.get('sort') as keyof Person;
+    const isReversed = searchParams.get('order') === 'desc';
+    const query = searchParams.get('query');
+    const sexFilter = searchParams.get('sex');
+    const centuries = searchParams.getAll('centuries');
+
+    const visiblePeople = useMemo(() => (
+      getFilteredPeople(people, query, sexFilter, centuries)
+    ), [people, query, sexFilter, centuries]);
+
+    const sortedPeople = useMemo(() => (
+      getSortedPeople(visiblePeople, sort, isReversed)
+    ), [visiblePeople, sort, isReversed]);
+
+    if (!sortedPeople.length) {
+      return <p>There are no people matching the current search criteria</p>;
+    }
+
+    const updateSearchParams = (sortBy: SortBy) => {
```

Comment:
will be better to move it to some helper function instead of create this function inside the component